app/models/Notification.py

```diff
@@ -0,0 +1,140 @@
+from app.instances.db import db
+from app.notifications import get_title
+from app.notifications import get_body
+from app.helpers.SerializableEnum import SerializableEnum
+from uuid import UUID
+from M2Crypto.m2 import rand_bytes
+import datetime
+
+class NotificationType(SerializableEnum):
+    STATUS_UPDATE = 0
+    NEW_ANSWER = 1
+    OUTGOLFED = 2
+    NEW_POST_COMMENT = 3
+    NEW_ANSWER_COMMENT = 4
+    ANSWER_VOTE = 5
+    POST_VOTE = 6
+
+class NotificationStatus(SerializableEnum):
+    UNSEEN = 0
+    SEEN = 1
+    READ = 2
+
+class Notification(db.Model):
+    """
+    Represents a notification sent to a user
+    """
+
+    __tablename__ = 'notifications'
+    __table_args__ = {'extend_existing': True}
+
+    id = db.Column(db.String(36), primary_key=True, default=lambda: str(UUID(bytes=rand_bytes(16))))
```

Comment:
Random primary keys will absolutely kill performance in InnoDB - the table size will be considerably larger, and lookup performance by PK will be much worse. Use an auto increment integer as the ID, and store this as an indexed `db.Binary` column to conserve space.